builtin/logical/pki/path_fetch.go

```diff
@@ -315,6 +333,79 @@ reply:
 	return
 }
 
+func (b *backend) readCertByAlias(ctx context.Context, req *logical.Request, alias string) ([]byte, error) {
+	// ca_chain as an alias needs special handling.
+	if alias == "ca_chain" {
+		caInfo, err := fetchCAInfo(ctx, b, req)
+		if err != nil {
+			return nil, err
+		}
+
+		caChain := caInfo.GetCAChain()
+		var certStr string
+		for _, ca := range caChain {
+			block := pem.Block{
+				Type:  "CERTIFICATE",
+				Bytes: ca.Bytes,
+			}
+			certStr = strings.Join([]string{certStr, strings.TrimSpace(string(pem.EncodeToMemory(&block)))}, "\n")
+		}
+
+		return []byte(strings.TrimSpace(certStr)), nil
+	}
+
+	certEntry, err := fetchCertBySerial(ctx, req, req.Path, alias)
+	if err != nil {
+		return nil, err
+	}
+	if certEntry == nil {
+		return nil, nil
+	}
+
+	return certEntry.Value, nil
+}
+
+func (b *backend) pathFetchReadRaw(ctx context.Context, req *logical.Request, data *framework.FieldData, serial string, pemType string, contentType string) (response *logical.Response, retErr error) {
+	var certificate []byte
+
+	// Errors don't cause an immediate exit, because the raw
+	// paths still need to return raw output, even if it is empty.
+	// The error will instead be logged (in the event the log surfaces
+	// warnings).
+	certificate, retErr = b.readCertByAlias(ctx, req, serial)
+	if retErr != nil {
+		goto reply
+	}
+
+	if len(pemType) != 0 {
+		certificate = marshalPem(pemType, certificate)
+	}
+
+reply:
+	response = &logical.Response{
+		Data: map[string]interface{}{
+			logical.HTTPContentType: contentType,
+			logical.HTTPRawBody:     certificate,
+		},
+	}
+
+	if retErr != nil {
+		if b.Logger().IsWarn() {
+			b.Logger().Warn("possible error, but cannot return in raw response. Note that an empty CA probably means none was configured, and an empty CRL is possibly correct; call /crl/rotate to create a non-empty CRL", "error", retErr)
```

Comment:
Could we not use this to set a non 2** status code?

Reply:
Oh yeah, that'd be a good idea.